src/bin/sage-fixdoctests

```diff
@@ -14,93 +14,271 @@ AUTHORS::
   situations when either the expected output or computed output are empty.
   Added doctest to sage.tests.cmdline
 """
+
+# ****************************************************************************
+#       Copyright (C) 2006 William Stein
+#                     2009 Nicolas M. Thiery
+#                     2013 Andrew Mathas
+#                     2014 Volker Braun
+#                     2020 Jonathan Kliem
+#                     2021 Frédéric Chapoton
+#                     2023 Matthias Koeppe
+#
+#  Distributed under the terms of the GNU General Public License (GPL)
+#  as published by the Free Software Foundation; either version 2 of
+#  the License, or (at your option) any later version.
+#                  https://www.gnu.org/licenses/
+# ****************************************************************************
+
+import itertools
 import os
+import re
+import shlex
 import subprocess
+import sys
+
 from argparse import ArgumentParser, FileType
+from pathlib import Path
+
+from sage.doctest.control import skipfile
+from sage.doctest.parsing import parse_file_optional_tags, parse_optional_tags, unparse_optional_tags, update_optional_tags
+from sage.env import SAGE_ROOT
+from sage.features import PythonModule
+from sage.features.all import all_features, module_feature, name_feature
 from sage.misc.temporary_file import tmp_filename
 
 parser = ArgumentParser(description="Given an input file with doctests, this creates a modified file that passes the doctests (modulo any raised exceptions). By default, the input file is modified. You can also name an output file.")
 parser.add_argument('-l', '--long',
                     dest='long', action="store_true", default=False)
-parser.add_argument("input", help="input filename",
-                    type=FileType('r'))
-parser.add_argument('output', nargs='?', help="output filename",
-                    type=FileType('w'))
+parser.add_argument("--distribution", type=str, default='',
+                    help="distribution package to test, e.g., 'sagemath-graphs', 'sagemath-combinat[modules]'; sets defaults for --venv and --environment")
+parser.add_argument("--venv", type=str, default='',
+                    help="directory name of a venv where 'sage -t' is to be run")
+parser.add_argument("--environment", type=str, default='',
+                    help="name of a module that provides the global environment for tests; implies --keep-both and --full-tracebacks")
+parser.add_argument("--no-test", default=False, action="store_true",
+                    help="do not run the doctester, only rewrite '# optional/needs' tags; implies --only-tags")
+parser.add_argument("--full-tracebacks", default=False, action="store_true",
```

Comment:
This says ```'# optional/needs' tags```, but I think we should stick to one choice, namely "needs". Technically we allow both, but we should guide people to use "needs" rather than "optional", unless there are cases where "optional" is more approprite.

Reply:
Ah, here it says "rewrite # optional/needs tags". As there are already "optional" tags, mentioning "optional" is right. OK.

Reply:
The guidance is here: https://deploy-preview-35749--sagemath-tobias.netlify.app/developer/coding_basics.html#special-markup-to-influence-doctests

Reply:
The fixer rewrites it so that `# optional` is used for optional packages/features, and `# needs` is used for standard packages/features

Reply:
Good. OK.